### PR TITLE
Allow passing checked items as a prop to Table

### PR DESCRIPTION
**Pull request by sradevski**
Change-type: minor
Signed-off-by: Stevche Radevski <stevche@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -901,6 +901,7 @@ cause a re-render of the component and will not be reflected on the table.
 | ------------- | --------- | --------- | ---------- | ---------------------------------------------------- |
 | `columns`    | `object[]` | - | ✓ | An array of column objects, as described above |
 | `data`    | `T[]` | - | ✓ | An array of objects that will be displayed in the table |
+| `checkedItems`    | `T[]` | - | ✓ | An array of objects that controls which rows are selected |
```
Comment (StefKors) [inline README.md:904]: Should this be required without a default?

File: src/components/Table/index.tsx [hunk 2]
```diff
@@ -578,6 +583,7 @@ export interface TableSortOptions<T> {
 export interface TableProps<T> {
 	columns: Array<TableColumn<T>>;
 	data?: T[] | null;
+	checkedItems?: T[];
```
Comment (StefKors) [inline src/components/Table/index.tsx:586]: This indicates optional

**Review by StefKors — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sradevski) [reply README.md:904]: @StefKors the docs are wrong, it should be optional.

**Review by LucianBuzzo — APPROVED** [review]
LGTM

**Review by StefKors — APPROVED** [review]

**Comment by sradevski** [comment]
@balena-ci retest

**Pull request merged**